test/unit/stats.js

```diff
@@ -0,0 +1,793 @@
+        ' negative)');
+
+        // assert.equal(err.name, 'InvalidArgumentError');
+        // assert.equal(err.propertyName, 'stat_id');
+        // assert.equal(err.propertyValue, req.params.stat_id);
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    xit('should give an error if the stat_id is not in the database',
+    function(done) {
+      var req = {
+        params: {
+          stat_id: 9999999
+        }
+      };
+
+      var promise = stats.getStat(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given stat_id is not found in the database');
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+  });
+
+  describe('createStat(req)', function() {
+    // Post a new row of stats into stats database
+    xit('should add new stats to the database', function(done) {
+      var req = {
+        body: {
+          student_id: 2,
+          event_id: 2,
+          height: 320,
+          weight: 54,
+          pacer: 382,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStat(req);
+      })
+      .then(function() {
+        return stats.getStats({});
+      })
+      .then(function(data) {
+        assert.lengthOf(data, statCount + 1);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6, fakeStat7]);
+        done();
+      });
+    });
+
+    // TODO Insert test for trying to post if stat already exists
+    // (should do nothing. Make sure of that.)
+
+    // Attempt to post existing stat
+    xit('adding an existant set of stat should do nothing', function(done) {
+      var req = {
+        body: {
+          student_id: 1,
+          event_id: 1,
+          height: 5,
+          weight: 5,
+          pacer: 5
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStat(req);
+      })
+      .then(function() {
+        return stats.getStats({});
+      })
+      .then(function(data) {
+        assert.lengthOf(data, statCount);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        assert.equal(result.message,
+        'Unable to add student because this set of stats already exists');
+        done();
+      });
+    });
+
+    // Attempt to post stats but missing body
+    xit('attempting to post stats with missing body will result in an error', function(done) {
+      var req = {
+        body: {}
```

Comment:
Body isn't missing here, the required fields **inside** body are. Change this test to say "attempting to post stats with missing fields will result in an error." Also remember to include params: {} inside your req. Basically, make sure one field or component is missing at a time, and test that there's an error.

E.g. The following test cases:

- Should not post stats if request is missing body
- Should not post stats if request is missing params section
- Should not post stats if params section is missing student_id or event_id
- Should not post stats if body section is missing height, weight or pacer (you already wrote these)

The perfect request looks like this: 
req = { 
params: {
event_id: #,
student_id: #
},
body {
height: #,
weight: ###,
pacer: ###
}
}

Write a test removing each piece. Check out the students test for reference. A lot of it is gonna be copy paste after you finish one. You've already written most of em, just fix the requests passed in to each test and be wary of what belongs in req.params vs req.body